flight/Libraries/math/coordinate_conversions.c

```diff
@@ -366,6 +366,75 @@ void rot_mult(float R[3][3], const float vec[3], float vec_out[3], bool transpos
 }
 
 /**
+ * @brief LLA_linearization_float calculates linearized NED to LLA conversion using floats
+ * @param home_latitude is the home latitude times 10e6
+ * @param home_altitude
+ * @param T is the linearized NED to LLA conversion
+ */
+void LLA_linearization_float(int32_t home_latitude_1e7, float home_altitude, float T[])
+{
+	float lat_rad = home_latitude_1e7 / 10.0e6f * DEG2RAD;
+	float alt = home_altitude;
+
+	// T is the linearization factor from LLA [rad] to NED [meters] divided by a scaling factor of
+	// 1e7*DEG2RAD. By default, input units are stored as degrees times 1e7 to avoid precision loss.
+	T[0] = ((alt + WGS84_A*1000.0f) / 10e6f) * DEG2RAD;
+	T[1] = ((cosf(lat_rad)*(alt + WGS84_A*1000.0f)) / 10e6f) * DEG2RAD ;
+	T[2] = -1.0f;
+}
+
+/**
+ * @brief LLA_linearization_double calculates linearized NED to LLA conversion using doubles (used in simulation)
+ * @param home_latitude
+ * @param home_altitude
+ * @param T is the linearized NED to LLA conversion
+ */
+void LLA_linearization_double(int32_t home_latitude_1e7, double home_altitude, double T[])
+{
+	double lat_rad = home_latitude_1e7 / 10.0e6 * ((double)DEG2RAD);
+	double alt = home_altitude;
+
+	// T is the linearization factor from LLA [rad] to NED [meters] divided by a scaling factor of
+	// 1e7*DEG2RAD. By default, input units are stored as degrees times 1e7 to avoid precision loss.
+	T[0] = ((alt + ((double)WGS84_A)*1000.0) / 10.0e6) * ((double)DEG2RAD);
+	T[1] = ((cos(lat_rad)*(alt + ((double)WGS84_A)*1000.0)) / 10.0e6) * ((double)DEG2RAD);
+	T[2] = -1.0;
+}
+
+/**
+ * @brief get_linearized_3D_transformation Map position from one coordinate system to
+ * another, linearized around the origin in the original coordinate system. For instance,
+ * this can be used to map from Lattitude-Longitude-Altitude to North-East-Down.
+ *
+ * Note that the scale of the input units is not important, as that is included in
+ * the mapping coefficients. For example, if we use decimal degrees * 1e7, then the
+ * mapping coefficient includes a division by 1e7.
+ *
+ * Current position:
+ * @param [in] theta_prime (typically latitude * 1e7), as an int32_t
+ * @param [in] phi_prime   (typically longitude * 1e7), as an int32_t
+ * @param [in] r_prime     (typically altitude), as a float
+ * Home position:
+ * @param [in] theta
+ * @param [in] phi
+ * @param [in] r
+ * Scale factors and outputs:
+ * @param [in] C mapping coefficient
+ * @param [out] XYZ output coordinates
+ */
+void get_linearized_3D_transformation(const int32_t theta_prime, const int32_t phi_prime, const float r_prime,
```

Comment:
https://github.com/TauLabs/TauLabs/blob/next/flight/Modules/GPS/UBX.c#L206 is what I was thinking of getting rid of. The native input to the system is in [mm] so I'm proposing we could remove the conversion to `float` and then the rest follows naturally. Was that clear?

On the other hand, we've been playing around with RTK GPS and sub-mm level precisioning is already something we're having fun with. So it might not make a ton of sense to optimize this for older tech GPSes when we really should be aiming at supporting an order of magnitude more precision in the near future.